### PR TITLE
tests: kernel: fp_sharing: Added support for Cortex-M7

### PR DESCRIPTION
**Pull request by AdithyaBaglody**
Added required macros to get the test case working with a
cortex-M7.

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>

### CODE REVIEW

File: tests/kernel/fp_sharing/src/float_regs_arm_gcc.h [hunk 1]
```diff
@@ -12,8 +12,8 @@
 #ifndef _FLOAT_REGS_ARM_GCC_H
 #define _FLOAT_REGS_ARM_GCC_H
 
-#if !defined(__GNUC__) || !defined(CONFIG_CPU_CORTEX_M4)
-#error __FILE__ goes only with Cortex-M4 GCC
+#if !defined(__GNUC__) || !(defined(CONFIG_CPU_CORTEX_M4) || defined(CONFIG_CPU_CORTEX_M7))
```
Comment (galak) [inline tests/kernel/fp_sharing/src/float_regs_arm_gcc.h]: Should this be !defined(CONFIG_CPU_CORTEX_M7)?

**Review by galak — DISMISSED** [review]
Rather than doing these based on CORTEX_M4/M7, how about changing the defines to something like CONFIG_ARMV7_M_ARMV8_M_FP?

**Review by nashif — APPROVED** [review]

File: tests/kernel/fp_sharing/testcase.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 tests:
   kernel.fp_sharing:
     extra_args: PI_NUM_ITERATIONS=70000
-    platform_whitelist: frdm_k64f
+    platform_whitelist: frdm_k64f sam_e70_xplained
```
Comment (galak) [inline tests/kernel/fp_sharing/testcase.yaml]: I wondering if this should be based on a CONFIG option instead of explicit platform choice.

**Review by galak — COMMENTED** [review]
Looks good, one comment about how we select the boards/platforms to run this.  Wondering if we should do filter: CONFIG_ARMV7_M_ARMV8_M_FP

**Comment by nashif** [comment]
> Looks good, one comment about how we select the boards/platforms to run this. Wondering if we should do filter: CONFIG_ARMV7_M_ARMV8_M_FP

makes sense

**Pull request merged**